control-plane/subcommand/server-acl-init/command.go

```diff
@@ -804,6 +758,73 @@ func (c *Command) createAuthMethod(consulClient *api.Client, authMethod *api.ACL
 		})
 }
 
+type gatewayRulesGenerator func(name, namespace string) (string, error)
+
+func (c *Command) configureGateway(gatewayType string, gatewayNames []string,
+	authMethodName string, ruleGenerator gatewayRulesGenerator, consulDC string,
+	primaryDC string, primary bool, consulClient *api.Client) error {
```

Comment:
can we move some of the gateway-specific params here into a `GatewayConfig` struct that we can pass into this method to simplify the params?